synapse/http/client.py

```diff
@@ -726,6 +726,14 @@ async def get_file(
                 HTTPStatus.BAD_GATEWAY, "Got error %d" % (response.code,), Codes.UNKNOWN
             )
 
+        if b"Content-Type" in resp_headers:
```

Comment:
Unfortunately `get_file` is meant to be a generic utility, not specific to URL previews, so I don't think putting this check here makes much sense. Maybe passing in a list of allowed (or not allowed) prefixes would work though?

Reply:
Ah right, this is used for media downloads as well. How about passing a predicate function rather than a list? That makes it easier to encode validating by prefix and other possible behaviours.

Reply:
Yep, that sounds reasonable to me!